File: codeforces/271/A.java
/******************************************************************************

Welcome to GDB Online.
GDB online is an online compiler and debugger tool for C, C++, Python, Java, PHP, Ruby, Perl,
C#, VB, Swift, Pascal, Fortran, Haskell, Objective-C, Assembly, HTML, CSS, JS, SQLite, Prolog.
Code, Compile, Run and Debug online from anywhere in world.

*******************************************************************************/
import java.util.*;
public class Main
{
	public static void main(String[] args) {
	    Scanner scan=new Scanner(System.in);
	int n=scan.nextInt();String str="";
	while(true){
	    n++;
	    str=Integer.toString(n);
	    if(str.charAt(0)!=str.charAt(1)&&str.charAt(0)!=str.charAt(2)&&str.charAt(0)!=str.charAt(3)&&str.charAt(1)!=str.charAt(2)&&str.charAt(1)!=str.charAt(3)&&str.charAt(2)!=str.charAt(3)){
	        
	        System.out.println(n);return;
	    }
	}
}


}

File: codeforces/271/A.cpp
/******************************************************************************
 
Welcome to GDB Online.
GDB online is an online compiler and debugger tool for C, C++, Python, Java, PHP, Ruby, Perl,
C#, OCaml, VB, Swift, Pascal, Fortran, Haskell, Objective-C, Assembly, HTML, CSS, JS, SQLite, Prolog.
Code, Compile, Run and Debug online from anywhere in world.
 
*******************************************************************************/
#include <iostream>
#include<map>
#include<algorithm>
#include<cstring>
#include<cmath>
 
using namespace std;                                          
 
int main(){
int x;
cin>>x;
x++;
while(x%10==(x/10)%10||x%10==(x/100)%10||x%10==(x/1000)%10||(x/10)%10==(x/100)%10||(x/10)%10==(x/1000)%10||(x/100)%10==(x/1000)%10){
    x++;
}
   cout<<x;
    return 0;
}

